src/mobster/cmd/generate/oci_image/spdx_utils.py

```diff
@@ -0,0 +1,317 @@
+"""SPDX-2.X utilities for the generate oci-image target"""
+
+import json
+import uuid
+from datetime import datetime, timezone
+from typing import Any
+
+from spdx_tools.spdx.model.actor import Actor, ActorType
+from spdx_tools.spdx.model.annotation import Annotation, AnnotationType
+from spdx_tools.spdx.model.checksum import Checksum, ChecksumAlgorithm
+from spdx_tools.spdx.model.document import Document
+from spdx_tools.spdx.model.package import (
+    ExternalPackageRef,
+    ExternalPackageRefCategory,
+    Package,
+)
+from spdx_tools.spdx.model.relationship import Relationship, RelationshipType
+
+from mobster import get_mobster_version
+from mobster.image import Image
+from mobster.sbom.spdx import get_package
+
+BUILDER_IMAGE_PROPERTY = {
+    "name": "konflux:container:is_builder_image:additional_builder_image",
+    "value": "script-runner-image",
+}
+
+
+async def normalize_actor(actor: str) -> str:
+    """
+    Adds a necessary actor classificator if not present.
+    This allows the SPDX library to load the actor without
+    validation issues.
+    Defaults to `TOOL`.
+    Args:
+        actor (str): The input actor.
+    Returns:
+        str: The normalized actor.
+    """
+    if not actor.upper().startswith(("TOOL: ", "ORGANIZATION: ", "PERSON: ")):
+        return "Tool: " + actor
+    return actor
+
+
+async def normalize_package(package: dict[str, Any]) -> None:
+    """
+    Adds necessary fields to an SPDX Package to be loaded by the
+    SPDX library without validation issues.
+    Args:
+        package (dict[str, Any]): The package to be normalized.
+
+    Returns:
+        None: Nothing, changes are performed in-place.
+    """
+    if "downloadLocation" not in package:
+        package["downloadLocation"] = "NOASSERTION"
+    if "name" not in package:
+        package["name"] = ""
+    if supplier := package.get("supplier"):
+        package["supplier"] = await normalize_actor(supplier)
+
+
+async def normalize_sbom(sbom: dict[str, Any]) -> None:
+    """
+    Adds necessary fields to an SPDX SBOM to be loaded by the
+    SPDX library without validation issues.
+    Args:
+        sbom (dict[str, Any]): The SBOM to be normalized.
+
+    Returns:
+        None: Nothing, changes are performed in-place.
+    """
+    if "SPDXID" not in sbom:
+        sbom["SPDXID"] = "SPDXRef-DOCUMENT"
+    if "dataLicense" not in sbom:
+        sbom["dataLicense"] = "CC0-1.0"
+    if "spdxVersion" not in sbom:
+        sbom["spdxVersion"] = "SPDX-2.3"
+    if "documentNamespace" not in sbom:
+        sbom["documentNamespace"] = f"https://konflux-ci.dev/spdx/{uuid.uuid4()}"
+    if "name" not in sbom:
```

Comment:
I think you're missing a funtionality that sets a valid name based on provided image pullspec and its digest. This functionality is in add_image_reference.py script but I can't find it here and resulting sboms contains a name that is inherited from the syft sbom.